single_controller/ray/base.py

```diff
@@ -187,6 +187,9 @@ def __init__(self,
         self.ray_cls_with_init = ray_cls_with_init
         self.name_prefix = get_random_string(length=6) if name_prefix is None else name_prefix
 
+        if worker_names is not None:
+            self._worker_names = worker_names
```

Comment:
I wonder if adding `assert self._is_init_with_detached_workers` between L190-191 would be better?

Your modification will not cause errors in current usage but it may be better to add this line.

Reply:
updated!